src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/HyperlinkClickedTelemetryEvent.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Common;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.Telemetry
+{
+    public sealed class HyperlinkClickedTelemetryEvent : TelemetryEvent
+    {
+        internal const string HyperlinkClickedEventName = "HyperlinkClicked";
+        internal const string AlternativePackageIdPropertyName = "AlternativePackageId";
+        internal const string HyperLinkTypePropertyName = "HyperlinkType";
+        internal const string CurrentTabPropertyName = "CurrentTab";
+        internal const string IsSolutionViewPropertyName = "IsSolutionView";
+
+        public HyperlinkClickedTelemetryEvent(HyperlinkType hyperlinkType, ItemFilter currentTab, bool isSolutionView)
+            : base(HyperlinkClickedEventName)
+        {
+            base[HyperLinkTypePropertyName] = hyperlinkType;
+            base[CurrentTabPropertyName] = currentTab;
+            base[IsSolutionViewPropertyName] = isSolutionView;
+        }
+
+        public HyperlinkClickedTelemetryEvent(HyperlinkType hyperlinkType, ItemFilter currentTab, bool isSolutionView, string alternativePackageId)
+            : this(hyperlinkType, currentTab, isSolutionView)
+        {
+            AddPiiData(AlternativePackageIdPropertyName, VSTelemetryServiceUtility.NormalizePackageId(alternativePackageId));
```

Comment:
Should we add information about which source the package is from too? I realize this information is captured in previously emitted events, but, joins in Visual Studio telemetry are expensive on large data sets.

Reply:
Why it would be useful ? Which questions would you like to answer with feed information ?

Reply:
From my understanding the deprecation link will search for `packageid:X`, which may not work on all package sources. This telemetry would let us see if this problem is worth solving.

Is there any reasons to be conservative with what data we add to VS telemetry events? Given the slow release cadence, it seems better to add what may be useful instead of only what we know we need... but I don't know VS's best practices guideline in this area :)

Reply:
Sounds good.

I just want to make sure I'm using the right amount of telemetry, and, I have planned a second iteration of this telemetry for next VS Preview release.

Also, deprecation metadata is gathered from the active feed the user has selected in PM UI. In the case of 'All' sources, the first NuGet feed/source that replies with deprecation metadata is shown to the user. We currently don't track deprecation metadata source feed origin. So, there are two options:

- either we just track current selected feed + wheter the all sources contains a non-nuget.org remote feed (most feasible, from my point of view)
- Make the whole metadata tracking from feed

I will add this telemetry in https://github.com/NuGet/Client.Engineering/issues/1124